opentelemetry-otlp/src/span.rs

```diff
@@ -159,4 +159,12 @@ impl opentelemetry_sdk::trace::SpanExporter for SpanExporter {
             SupportedTransportClient::Http(client) => client.set_resource(resource),
         }
     }
+    fn shutdown(&self) -> OTelSdkResult {
```

Comment:
Hey @shivanshuraj1333 thanks for the interest in contributing! 

Looks like you need:

```suggestion
    fn shutdown(&mut self) -> OTelSdkResult {
```

Reply:
depending on the scope of the PR this needs much more changes to actually make `shutdown()` work without the need of mutability or this can be tackled via a follow-up PR. @cijothomas thoughts, because you created the issue? 😅

Reply:
yes this might be more involved. Following what Log signal is doing would be a good start.

Reply:
Can I get a one liner description of what is the expected output of the `shutdown` implementation of traces?
How can I decide if the self is mutable or not? Following the changes from https://github.com/open-telemetry/opentelemetry-rust/issues/2770, I need some inputs on how to test it end to end, if the shutdown for traces working as expected.

I'm new to rust, that's why some questions, to get started.

Reply:
update: have made the necessary changes